splunk_eventgen/lib/eventgentimer.py

```diff
@@ -187,12 +187,11 @@ def real_run(self):
                             for worker_id in range(self.config.generatorWorkers):
                                 # self.generatorPlugin is only an instance, now we need a real plugin. Make a copy of
                                 # of the sample in case another generator corrupts it.
-                                copy_sample = copy.copy(self.sample)
```

Comment:
I am not sure why we need to make a copy of the example here since for each sample we put a generator object into queue and a  work will pick it up and process it.  Previous fix (https://github.com/splunk/eventgen/pull/162/files) has great performance downgrade for multiprocess mode.

Reply:
Think this is more from legacy.  I'm guessing this comes from the old eg 5 days.  Think I had to do this when sample's weren't fully loaded in mem, or you'd have 1 sample save a file into a new sample.